hooks/command

```diff
@@ -209,6 +209,9 @@ fi
 if [[ -z "$status" ]]; then
   echo "Warning: could not get actual exit code of the job"
   status="42"
+elif [[ "$status" == "null" ]]; then
```

Comment:
Thanks for catching this (again :) I remember this popped up in a previous PR and realised I missed that we didn't follow up on it https://github.com/EmbarkStudios/k8s-buildkite-plugin/pull/14/files#r503501353
Can we use the proposed solution there instead to have jq exit with non-zero if it doesn't find a status?

Reply:
Happy to go down that path, but what do you think we should do in the case there is an error from jq? Do you think we should keep looping? I imagine not because I don't think we'll ever expect to get a status if there isn't one initially (I must be missing something otherwise I don't think we'd loop at all) and I don't think we'd want to fallback to any other status since that would ignore the fact that this one seemed to fail

Reply:
Yeah I think it should be alright to keep looping - we'll retry and eventually exit the loop if jq fails for some unrecoverable reason, status will be left unset and we'll hit https://github.com/EmbarkStudios/k8s-buildkite-plugin/blob/ef26ce030c8bb183cc34c904b43fac5d43bf213f/hooks/command#L209

The retry is mostly for things like connectivity issues, we could try to infer whether or not jq is failing for some unrecoverable reason (e.g if it didn't get a json in the first place because the pod didn't exist) and try to bail out early but there doesn't seem to be a need to

Reply:
Updated to make jq error in the case it's missing, the only thing I'm not sure on is the looping question above still. Also I didn't do that for the init container case since otherwise it would fail instantly if that was unfetchable, in which case there might as well not even be a loop?